src/sas/qtgui/Perspectives/Invariant/InvariantDetails.py

```diff
@@ -77,13 +78,29 @@ def showDialog(self):
         # Pull out data from the model
         self.qstar_total = float(self._model.item(WIDGETS.W_INVARIANT).text())
 
-        self.txtQData.setText(str(self.qstar_total))
-        self.txtQDataErr.setText(self._model.item(WIDGETS.W_INVARIANT_ERR).text())
-
         # Reset progress counters
         self.progress_low_qstar = 0.0
         self.progress_high_qstar = 0.0
-        self.progress_qstar = 100.0
+        self.progress_data_qstar = 100.0
+
+        # Reset numerical values
+        self.txtQData.setText(None)
+        self.txtQDataErr.setText(None)
+        self.txtQLowQ.setText(None)
+        self.txtQLowQErr.setText(None)
+        self.txtQHighQ.setText(None)
+        self.txtQHighQErr.setText(None)
+
+
+        # Q* from data
+        self.qdata = float(self._model.item(WIDGETS.D_DATA_QSTAR).text())
+
+        self.txtQData.setText(str(self.qdata))
+        self.txtQDataErr.setText(self._model.item(WIDGETS.D_DATA_QSTAR_ERR).text())
+        try:
+            self.progress_data_qstar = (self.qdata/self.qstar_total)*100.0
+        except:
```

Comment:
Again am going to ask to defer this.  This would be a  change to the original structure of the code.  The change here is another copy paste from the other identical sections below it. I would rather keep the style the same until we change all of them. Otherwise it becomes hard to follow (at least for me who is not a python ninja 😃  when each related bit is written with a different style.